c7n/resources/asg.py

```diff
@@ -509,26 +520,27 @@ def get_unencrypted_configs(self, ec2):
                     for bd in image['BlockDeviceMappings'] if 'Ebs' in bd}
             else:
                 image_block_devs = {}
-            for bd in c['BlockDeviceMappings']:
-                if 'Ebs' not in bd:
-                    continue
-                # Launch configs can shadow image devices, images have
-                # precedence.
-                if bd['DeviceName'] in image_block_devs:
-                    continue
-                if 'SnapshotId' in bd['Ebs']:
-                    snaps.setdefault(
-                        bd['Ebs']['SnapshotId'].strip(), []).append(cid)
-                elif not bd['Ebs'].get('Encrypted'):
-                    unencrypted_configs.add(cid)
+            if 'BlockDeviceMappings' in c:
+                for bd in c['BlockDeviceMappings']:
```

Comment:
`c.get('BlockDeviceMappings', ()):` is a little nicer and let's you dedent this block.

Reply:
added